pkgmgr/package.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 Blink Labs Software
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pkgmgr
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+type Package struct {
+	Name         string
+	Version      string
+	Description  string
+	InstallSteps []PackageInstallStep
+}
+
+func (p Package) install(cfg Config) error {
+	pkgName := fmt.Sprintf("%s-%s", p.Name, p.Version)
+	for _, installStep := range p.InstallSteps {
+		if installStep.Docker != nil {
+			if err := installStep.Docker.install(cfg, pkgName); err != nil {
+				return err
+			}
+		} else if installStep.File != nil {
```

Comment:
File before Docker.

Reply:
This is specified per install step, so the order of this if/else doesn't matter. I added some comments and an additional check to prevent specifying multiple install methods on a single install step.